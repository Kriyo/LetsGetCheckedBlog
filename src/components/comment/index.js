import React from 'react'

import {
  Container,
  Item,
  RootItem,
  NestedItem,
  Count,
  Image,
  DetailsWrapper,
  Content,
  AuthorDetails,
  ReplyHeading,
  Form,
  Button,
  ButtonWrapper,
  ErrorLabel,
  Input,
  TextArea,
  Section,
} from './styles/comment'

export const Comment = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Comment.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Comment.Section = ({ children, ...props }) => {
  return <Section {...props}>{children}</Section>
}

Comment.RootComment = ({ children, ...props }) => {
  return <RootItem {...props}>{children}</RootItem>
}

Comment.NestedComment = ({ children, ...props }) => {
  return <NestedItem {...props}>{children}</NestedItem>
}

Comment.Count = ({ children, ...props }) => {
  return <Count {...props}>{children}</Count>
}

Comment.DetailsWrapper = ({ children, ...props }) => {
  return <DetailsWrapper {...props}>{children}</DetailsWrapper>
}

Comment.AuthorDetails = ({ children, ...props }) => {
  return <AuthorDetails {...props}>{children}</AuthorDetails>
}

Comment.Avatar = ({ children, ...props }) => {
  return <Image {...props}>{children}</Image>
}

Comment.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}

Comment.ReplyHeading = ({ children, ...props }) => {
  return <ReplyHeading {...props}>{children}</ReplyHeading>
}

Comment.Form = ({ children, ...props }) => {
  return <Form {...props}>{children}</Form>
}

Comment.ButtonWrapper = ({ children, ...props }) => {
  return <ButtonWrapper {...props}>{children}</ButtonWrapper>
}

Comment.Button = ({ children, ...props }) => {
  return <Button {...props}>{children}</Button>
}
Comment.ErrorLabel = ({ children, ...props }) => {
  return <ErrorLabel {...props}>{children}</ErrorLabel>
}
Comment.UserInput = ({ children, ...props }) => {
  return <Input {...props}>{children}</Input>
}

Comment.TextArea = ({ children, ...props }) => {
  return <TextArea {...props}>{children}</TextArea>
}
