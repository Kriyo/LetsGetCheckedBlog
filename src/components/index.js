export { Comment } from './comment'
export { Feature } from './feature'
export { NavBar } from './navbar'
export { Post } from './post'
