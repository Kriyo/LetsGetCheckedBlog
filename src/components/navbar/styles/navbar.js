import styled from 'styled-components/macro'
import { Neutral } from '../../../utils/colors'

export const Background = styled.div`
  display: flex;
  flex-direction: column;
  background: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0.5),
      rgba(0, 0, 0, 0.75),
      rgba(0, 0, 0, 0.5)
    ),
    url('../images/blog-background.jpg') top left / cover no-repeat;
`

export const Container = styled.nav`
  display: flex;
  margin: 0 56px;
  height: 100px;
  justify-content: space-between;
  align-items: center;
  a {
    display: flex;
  }
  @media (max-width: 1000px) {
    margin: 0 30px;
  }
`

export const Link = styled.p`
  color: ${Neutral[100]};
  text-decoration: none;
  margin-right: 30px;
  font-weight: 'normal';
  cursor: pointer;
  &:hover {
    text-decoration: underline;
  }
`

export const Logo = styled.img`
  height: 50px;
  width: 50px;
  margin-right: 40px;
`
