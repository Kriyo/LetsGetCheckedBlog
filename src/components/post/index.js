import React from 'react'

import {
  Container,
  Item,
  Author,
  Title,
  Description,
  PublishDate,
  Content,
} from './styles/post'

export const Post = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Post.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Post.Author = ({ children, ...props }) => {
  return <Author {...props}>{children}</Author>
}

Post.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Post.Description = ({ children, ...props }) => {
  return <Description {...props}>{children}</Description>
}

Post.PublishDate = ({ children, ...props }) => {
  return <PublishDate {...props}>{children}</PublishDate>
}

Post.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}
