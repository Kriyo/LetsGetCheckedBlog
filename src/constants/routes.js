export const HOME = '/'
export const BLOG = '/blog'
export const BLOGPOST = '/blog/:slug'
export const ABOUT = '/about'
export const CONTACT = '/contact'
