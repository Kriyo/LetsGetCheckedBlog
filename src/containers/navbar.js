import React from 'react'
import { Nav } from '../components'
import logo from '../white-logo.png'

export function NavContainer({ children }) {
  return (
    <Nav>
      <Nav.Frame>
        <Nav.Logo alt="Lets Get Checked Blog Logo" src={logo} />
        <Nav.Link>Home</Nav.Link>
        <Nav.Link>Blog</Nav.Link>
        <Nav.Link>Contact</Nav.Link>
        <Nav.Link>About</Nav.Link>
      </Nav.Frame>
      {children}
    </Nav>
  )
}
