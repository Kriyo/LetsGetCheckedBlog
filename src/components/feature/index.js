import React from 'react'

import { Container, Title, SubTitle } from './styles/feature'

export const Feature = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Feature.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Feature.SubTitle = ({ children, ...props }) => {
  return <SubTitle {...props}>{children}</SubTitle>
}
