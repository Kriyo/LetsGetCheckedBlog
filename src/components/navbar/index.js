import React from 'react'
import { Background, Container, Link, Logo } from './styles/navbar'

export default function NavBar({ background = true, children, ...props }) {
  return background ? <Background {...props}>{children}</Background> : children
}

NavBar.Frame = function NavBarFrame({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

NavBar.Link = function NavBarLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>
}

NavBar.Logo = function NavBarLogo({ to, ...restProps }) {
  return <Logo {...restProps} />
}
