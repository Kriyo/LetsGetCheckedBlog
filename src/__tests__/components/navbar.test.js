import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { NavBar } from '../../components/navbar'
import { HOME, BLOG } from '../../constants/routes'

describe('<NavBar />', () => {
  it('renders the <Navbar /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Router>
        <NavBar>
          <NavBar.Frame>
            <NavBar.Logo to={HOME} src="/logo.svg" alt="Blog Logo" />
            <NavBar.Link alt="a link" to={BLOG}>
              I am a link
            </NavBar.Link>
          </NavBar.Frame>
        </NavBar>
      </Router>
    )

    expect(getByText('I am a link')).toBeTruthy()
    expect(getByTestId('nav-bg')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the <Navbar /> without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Router>
        <NavBar background={false}>
          <NavBar.Frame>
            <NavBar.Logo to={HOME} src="/logo.svg" alt="Blog Logo" />
            <NavBar.Link to={BLOG} alt="a link">
              I am another link
            </NavBar.Link>
          </NavBar.Frame>
        </NavBar>
      </Router>
    )

    expect(getByText('I am another link')).toBeTruthy()
    expect(queryByTestId('nav-bg')).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
