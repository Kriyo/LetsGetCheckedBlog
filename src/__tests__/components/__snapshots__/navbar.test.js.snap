// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders the <Navbar /> with a background 1`] = `
<div
  class="navbar__Background-sc-1waf4cx-0 fkfKQq"
  data-testid="nav-bg"
>
  <nav
    class="navbar__Container-sc-1waf4cx-1 iTnWlJ"
  >
    <a
      href="/"
    >
      <img
        alt="Blog Logo"
        class="navbar__Logo-sc-1waf4cx-3 dKtmPt"
        src="/logo.svg"
      />
    </a>
    <a
      alt="a link"
      class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
      href="/blog"
    >
      I am a link
    </a>
  </nav>
</div>
`;

exports[`<NavBar /> renders the <Navbar /> without a background 1`] = `
<nav
  class="navbar__Container-sc-1waf4cx-1 iTnWlJ"
>
  <a
    href="/"
  >
    <img
      alt="Blog Logo"
      class="navbar__Logo-sc-1waf4cx-3 dKtmPt"
      src="/logo.svg"
    />
  </a>
  <a
    alt="a link"
    class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
    href="/blog"
  >
    I am another link
  </a>
</nav>
`;
